.github/workflows/build-assets.yml

```diff
@@ -0,0 +1,161 @@
+# Invoking this pipeline requires additional permissions, so must be invoked
+# in a way to pass those permissions on, e.g.:
+#
+#  build-assets:
+#    permissions:
+#      contents: read
+#      id-token: write
+#      attestations: write
+#    uses: ./.github/workflows/build-assets.yml
+
+name: "Build the PIE assets"
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  build-phar:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system:
+          - ubuntu-latest
+        php-versions:
+          - '8.1'
+    permissions:
+      # id-token:write is required for build provenance attestation.
+      id-token: write
+      # attestations:write is required for build provenance attestation.
+      attestations: write
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: none
+          tools: composer, box
+          php-version: "${{ matrix.php-version }}"
+      - uses: actions/checkout@v6
+        with:
+          fetch-depth: 0
+      # Fixes `git describe` picking the wrong tag - see https://github.com/php/pie/issues/307
+      - run: git fetch --tags --force
+      # Ensure some kind of previous tag exists, otherwise box fails
+      - run: git describe --tags HEAD || git tag 0.0.0
+      - uses: ramsey/composer-install@v3
+      - name: Build PHAR
+        run: box compile
+      - name: Check the PHAR executes
+        run: php pie.phar --version
+      - name: Generate build provenance attestation
+        # It does not make sense to do this for PR builds, nor do contributors
+        # have permission to do. We can't write attestations to `php/pie` in an
+        # unprivileged context, otherwise anyone could send a PR with malicious
+        # code, which would store attestation that `php/pie` built the PHAR, and
+        # it would look genuine. So this should NOT run for PR builds.
+        if: github.event_name != 'pull_request'
+        uses: actions/attest-build-provenance@v3
+        with:
+          subject-path: '${{ github.workspace }}/pie.phar'
+      - uses: actions/upload-artifact@v5
+        with:
+          name: pie-${{ github.sha }}.phar
+          path: pie.phar
+
+  build-executable:
+    needs:
+     - build-phar
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system:
+          - ubuntu-24.04
+          - ubuntu-24.04-arm
+          - macos-15-intel
+          - macos-26
```

Comment:
OSX binaries may be problematic to provide; a test someone with an ARM64 Mac revealed:

<img width="241" height="292" alt="Image" src="https://github.com/user-attachments/assets/dd2c0a01-3f12-45c9-a9bb-06c78c923e2a" />

I assume we'll have to look at signing it somehow. Having never used a Mac in anger, I'll need to look into how we can do this in CI (if possible).

We could potentially leave it as is, at least for now; I expect if/when we get more adoption of the executable distribution, we can revisit this separately.

Reply:
Found this, may have useful references: https://github.com/simonw/datasette/issues/1171

Reply:
Yes you will need an Developer Account which costs 100$ a year at Apple. I setuped it 3 months ago for a Go project, is a pain. See https://github.com/anchore/quill

If you need help let me now, but you would need first an Paid Apple Developer Account 😅